src/plugins/teleport-plugin-react-styled-components/utils.ts

```diff
@@ -0,0 +1,31 @@
+import * as t from '@babel/types'
+import { cammelCaseToDashCase } from '../../shared/utils/string-utils'
+import { stringAsTemplateLiteral } from '../../shared/utils/ast-jsx-utils'
+
+export const generateStyledComponent = (name: string, type: string, styles: object) => {
+  return t.variableDeclaration('const', [
+    t.variableDeclarator(
+      t.identifier(name),
+      t.taggedTemplateExpression(
+        t.memberExpression(t.identifier('styled'), t.identifier(type)),
+        stringAsTemplateLiteral(mapStyles(styles))
+      )
+    ),
+  ])
+}
+
+const mapStyles = (styles: object) => {
```

Comment:
can we something from JSS to compile the style object into the style rules for a single component?

Reply:
No @alexnm I tried all possible ways that I am aware of, to do this job using JSS. But ended up writing this method 😅